resources/assets/lib/beatmap-discussions/event.tsx

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import BeatmapsetEventJson from 'interfaces/beatmapset-event-json';
+import UserJSON from 'interfaces/user-json';
+import { route } from 'laroute';
+import { Dictionary, kebabCase } from 'lodash';
+import * as React from 'react';
+import TimeWithTooltip from 'time-with-tooltip';
+
+interface Props {
+  discussions?: Dictionary<BeatmapDiscussion>;
+  event: BeatmapsetEventJson;
+  mode: 'discussions' | 'profile';
+  time?: string;
+  users: Dictionary<UserJSON>;
+}
+
+export default class Event extends React.PureComponent<Props> {
+  static readonly defaultProps = {
+    mode: 'discussions',
+  };
+
+  private get beatmapsetId(): number | undefined {
+    return this.props.event.beatmapset?.id;
+  }
+
+  private get discussionId(): number | undefined {
+    return this.props.event.comment?.beatmap_discussion_id;
+  }
+
+  private get discussion() {
+    return this.props.event.discussion ?? this.props.discussions?.[this.discussionId ?? ''];
+  }
+
+  private get firstPost() {
+    return this.discussion?.starting_post ?? this.discussion?.posts?.[0];
+  }
+
+  render() {
+    return this.props.mode === 'discussions'
+      ? this.renderDiscussionsVersion()
+      : this.renderProfileVersion();
+  }
+
+  renderDiscussionsVersion() {
+    const eventTime = this.props.time ?? this.props.event.created_at;
+
+    return (
+      <div className='beatmapset-event'>
+        <div className={osu.classWithModifiers('beatmapset-event__icon', [kebabCase(this.props.event.type)])} />
+        <div className='beatmapset-event__time'>
+          <TimeWithTooltip dateTime={eventTime} format='LT' />
+        </div>
+        <div
+          className={'beatmapset-event__content'}
+          dangerouslySetInnerHTML={{
+            __html: this.contentText(),
+          }}
+        />
+      </div>
+    );
+  }
+
+  renderProfileVersion() {
+    const cover = this.props.event.beatmapset?.covers?.list;
+    let discussionLink = route('beatmapsets.discussion', { beatmapset: this.beatmapsetId });
+    if (this.discussionId != null) {
+      discussionLink = `${discussionLink}#/${this.discussionId}`;
+    }
+
+    return (
+      <div className='beatmapset-event'>
+        <a href={discussionLink}>
+          <img className='beatmapset-activities__beatmapset-cover' src={cover} />
```

Comment:
I think this is time to do something with this random element without its block.